### DISCUSSION TITLE
Welcome to caddy-s3-proxy Discussions!

### DISCUSSION THREAD

**Discussion by rayjlinden** [lindenlab/caddy-s3-proxy#41] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by redturtlepower** [comment]
Hi!
I am about to publish some desktop applications, and need a download solution. My best (?) idea is to use caddy as a static file server, and let jenkins upload newly created installers.

Summarizing this comment, I have 2 questions, but first let me show what I imagine to be the end result.

Jenkins would upload files to the caddyserver 2 to a specific directory. Something along the lines:
```
curl -u jenkins:pass123 -Ffile=@app1-installer-v2.2.1.msi 'http://mydomain.com/upload?path=/apps/app1/2/2.2/app1-installer-v2.2.1.msi
```
Downloads then would look like:
```
curl -X GET http://mydomain.com/apps/app1/2/2.2/app1-installer-v2.2.1.msi
```
Now the questions.

1) I want to protect the /upload route via basicauth in my `Caddyfile`, so only my jenkins instance can post new files. This already seems to work and is not related to this project. **But can this plugin post/upload files to a specific directory as explained above?**

2) Moreover I like the idea of integrating s3. I am thinking about uploading the installers to a s3 bucket (as origin) instead of my own server. My users would still browse and download installers from my own server using caddy. The idea is: I would like this plugin to download the files from the s3 origin to the server that acts as a cache, so I don't have to pay for each download traffic from s3. I am not sure whether the current implementation caches s3 files, or requests/redirects them each time from s3. 
**Does this feature already exist or can it be configured? If not, can you imaging adding this feature?** (maybe I can also help...but I don't have experience with go)

For reference, here is my current Caddyfile:
```
:80

root * /var/www/cdn

file_server browse

@postupload {
  method POST
}
basicauth @postupload {
  jenkins JDJdasfgdsasdfafdsaasdffdsadsffdsadfsdfLzQvZkc2
}
```
